tests/gcm/test_auto.py

```diff
@@ -52,7 +52,9 @@ def test_given_linear_regression_problem_when_auto_assign_causal_models_with_goo
     data.update({"Y": Y})
 
     assign_causal_mechanisms(causal_model, pd.DataFrame(data), quality=AssignmentQuality.GOOD)
-    assert isinstance(causal_model.causal_mechanism("Y").prediction_model.sklearn_model, LinearRegression)
+    assert isinstance(
+        causal_model.causal_mechanism("Y").prediction_model.sklearn_model, LinearRegression
+    ) or isinstance(causal_model.causal_mechanism("Y").prediction_model.sklearn_model, Pipeline)
```

Comment:
Where does pipeline sneak in [here](https://github.com/py-why/dowhy/blob/main/dowhy/gcm/auto.py#L150)?

Reply:
Here: https://github.com/py-why/dowhy/blob/main/dowhy/gcm/ml/regression.py#L121
It adds polynomial features, which can be equally good as a linear model when the coefficients are zero accordingly.